crates/dekaf/src/task_manager.rs

```diff
@@ -99,23 +104,30 @@ impl TaskStateListener {
 pub struct TaskManager {
     // Key: materialization/task name
     tasks: std::sync::Mutex<
-        HashMap<String, std::sync::Weak<watch::Receiver<Option<TaskManagerResult<TaskState>>>>>,
+        HashMap<
+            String,
+            (
+                // Activity signal to keep the task manager alive
```

Comment:
@nit: I have a sense these mechanics could be simplified, but I'm not taking the time to come up with an actionable proposal, so :shrug: for now.

Reply:
Yeah... I agree it feels a bit circular. I'll keep thinking about simpler factorings